extensions/binding/org.eclipse.smarthome.binding.bosesoundtouch/src/main/java/org/eclipse/smarthome/binding/bosesoundtouch/internal/discovery/SoundTouchDiscoveryParticipant.java

```diff
@@ -52,20 +53,35 @@
     }
 
     @Override
+    @SuppressWarnings("deprecation")
     public DiscoveryResult createResult(ServiceInfo info) {
         DiscoveryResult result = null;
         ThingUID uid = getThingUID(info);
         if (uid != null) {
 
+            // remove the domain from the name
+            InetAddress[] addrs = info.getInetAddresses();
+
             Map<String, Object> properties = new HashMap<>(2);
             String label = "Bose SoundTouch";
             try {
-                label = info.getName();
+                if (BST_10_THING_TYPE_UID.equals(uid.getThingTypeUID())) {
+                    try {
+                        String group = DiscoveryUtil
+                                .executeUrl("http://" + addrs[0].getHostAddress() + ":8090/getGroup");
+                        label = DiscoveryUtil.getContentOfFirstElement(group, "name");
+                    } catch (Exception e) {
+                        logger.debug("Can't obtain label for group. Will use the default one");
+                        label = "";
+                    }
+                }
+
+                if (label == null || label.isEmpty()) {
+                    label = info.getName();
+                }
             } catch (Exception e) {

```

Comment:
where should this Exception originate from?

Reply:
In the original only `serviceInfo.getName()` was wrapped in the try/catch. But it also will not throw an exception. Neither does your new code I guess.

Reply:
> In the original only `serviceInfo.getName()` was wrapped in the try/catch.

That's correct, and I didn't want to change the original logic.